lib/email/receiver.rb

```diff
@@ -752,6 +753,16 @@ def send_subscription_mail(action, user)
       message = SubscriptionMailer.send(action, user)
       Email::Sender.new(message, :subscription).send
     end
+
+    def delete_staged_users
+      @staged_users.each do |user|
+        begin
+          UserDestroyer.new(Discourse.system_user).destroy(user, quiet: true)
+        rescue UserDestroyer::PostsExistError
```

Comment:
Instead of rescuing the error which is slow, maybe we can just add a filter in the `@staged_users` query to only return users without any posts?